src/linker/Linker.Steps/MarkExportedTypesTargetStep.cs

```diff
@@ -0,0 +1,40 @@
+// Licensed to the.NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class MarkExportedTypesTargetStep : BaseStep
+	{
+		protected override void ProcessAssembly (AssemblyDefinition assembly)
+		{
+			if (!assembly.MainModule.HasExportedTypes)
+				return;
+
+			foreach (var type in assembly.MainModule.ExportedTypes)
+				InitializeExportedType (type);
+		}
+
+		void InitializeExportedType (ExportedType exportedType)
+		{
+			if (!Annotations.IsMarked (exportedType))
+				return;
+
```

Comment:
Since this runs before `MarkStep` - what will happen if I specify descriptor like this?
```xml
<type fullname="MyExportedType" required="false" preserve="methods"/>
```

The exported type won't get marked by the descriptor (since `required=false`), and thus this step won't do anything. But when `MarkStep` actually marks it, it probably won't apply the `preserve=methods` to the target of the exported type, right? Should this be done in two parts - first handle the marked types (and mark the target) and then handle type preserve (and propagate it to the target regardless of marking)?

Reply:
It's OK to load the target assembly in this step to add the preserve onto the target type - so I would think that should work just fine. But test would be nice :-)

Reply:
Made extra tests to cover this and they pass